packages/adapter-components/src/elements/ducktype/transformer.ts

```diff
@@ -111,42 +132,167 @@ export const getTypeAndInstances = async ({
     log.debug(`storing full entries for ${type.elemID.name}`)
   }
 
-  const instances = await awu(naclEntries).flatMap(async (entry, index) => {
+  const idToInstances = Object.fromEntries((await awu(naclEntries).map(async (entry, index) => {
     if (nestedFieldDetails !== undefined) {
-      return awu(makeArray(entry[nestedFieldDetails.field.name])).map(
-        (nestedEntry, nesteIndex) => toInstance({
+      return [entry.id, await awu(makeArray(
+        entry.values.flatMap(value => value[nestedFieldDetails.field.name])
+      )).map(
+        async (nestedEntry, nesteIndex) => toInstance({
           entry: nestedEntry,
           type: nestedFieldDetails.type,
           transformationConfigByType,
           transformationDefaultConfig,
           defaultName: `unnamed_${index}_${nesteIndex}`, // TODO improve
           hasDynamicFields,
         })
-      ).filter(isDefined).toArray()
+      ).filter(isDefined).toArray()]
     }
 
-    return [await toInstance({
+    return [entry.id, [await toInstance({
       entry,
       type,
       transformationConfigByType,
       transformationDefaultConfig,
       defaultName: `unnamed_${index}`, // TODO improve
       hasDynamicFields,
-    })].filter(isDefined)
-  }).toArray()
+    })].filter(isDefined)]
+  }).toArray()) as [string, InstanceElement[]][])
 
-  if (type.isSettings && instances.length > 1) {
-    log.warn(`Expected one instance for singleton type: ${type.elemID.name} but received: ${instances.length}`)
+  const instanceElements = Object.values(idToInstances).flat()
+  if (type.isSettings && instanceElements.length > 1) {
+    log.warn(`Expected one instance for singleton type: ${type.elemID.name} but received: ${instanceElements.length}`)
     throw new Error(`Could not fetch type ${type.elemID.name}, singleton types should not have more than one instance`)
   }
 
-  const elements = [type, ...nestedTypes, ...instances]
+  const { recurseInto } = requestWithDefaults
+  const getExtraFieldValues = async (instances: InstanceElement[]):
+  Promise<Record<string, Record<string, Entries>>> => {
```

Comment:
indentation nit:
```suggestion
  const getExtraFieldValues = async (instances: InstanceElement[]): Promise<
    Record<string, Record<string, Entries>>
  > => {
```
or
```suggestion
  const getExtraFieldValues = async (
    instances: InstanceElement[]
  ): Promise<Record<string, Record<string, Entries>>> => {
```

(or just indent `Promise<...` more)